### PR TITLE
fix: update refit package to address security advisory

### PR DESCRIPTION
**Pull request by mahdichtioui**
GitHub Issue: #

## Proposed Changes

 - [x] Bug fix
 - [ ] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

Updating the Refit package version from 7.0.0 to 8.0.0 to address security advisory

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [ ] **Minor**
  - New functionalities were added.
- [x] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [x] Updated [CHANGELOG.md](../blob/main/CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [ ] Documentation files were updated according with the changes.
  - Update `README.md` and `TemplateConfig.md` if you made changes to **templating**.
  - Update `AzurePipelines.md` and `APP_README.md` if you made changes to **pipelines**.
  - Update `Diagnostics.md` if you made changes to **diagnostic tools**.
  - Update `Architecture.md` and its diagrams if you made **architecture decisions** or if you introduced new **recipes**.
  - ...and so forth: Make sure you update the documentation files associated to the recipes you changed. Review the topics by looking at the content of the `doc/` folder.
- [ ] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [ ] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [ ] Automated tests for the changes have been added/updated.
- [ ] Tested on all relevant platforms

### CODE REVIEW

**Comment by lamonfly** [comment]
This should be a major version change as it is a Fix!
Don't forget to go through the PR checklist. It should mention to bump the minor version and update the documentation.

**Comment by jeanplevesque** [comment]
> This should be a major version change as it is a Fix! Don't forget to go through the PR checklist. It should mention to bump the minor version and update the documentation.

@lamonfly @mahdichtioui  Actually, this repository is not publishing any nuget package that others can install as a dependency. It's an application template so it's closer to an application. In this context, the definition for Major, Minor, and Patch aren't the same (as described in the PR template). In this particular case, you should use a Patch with a `fix:` conventional commit.

File: src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj [hunk 1]
```diff
@@ -21,7 +21,7 @@
 		<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.1" />
 		<PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="3.1.0" />
 		<PackageReference Include="Reactive.Annex" Version="2.0.0" />
-		<PackageReference Include="Refit" Version="7.0.0" />
+		<PackageReference Include="Refit" Version="8.0.0" />
 		<PackageReference Include="ReviewService.Abstractions" Version="1.0.0" />
 		<PackageReference Include="System.Text.Json" Version="8.0.5" />
 		<PackageReference Include="MallardMessageHandlers" Version="1.2.0" />
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Access/ApplicationTemplate.Access.csproj:27]: You'll also want to update `MallardMessageHandlers` to 2.0.0 (which should be available soon now that your PR merged).

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
 Prefix your items with `(Template)` if the change is about the template and not the resulting application.
 
+## 4.0.X
+
+### Security
```
Comment (jeanplevesque) [inline CHANGELOG.md:10]: You don't need to add subheaders, nor notes. As you can see, we've already done security updates in the past (look at the mention of `System.Text.Json`).

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

**Review by jeanplevesque — APPROVED** [review]

**Review by Soap-141 — APPROVED** [review]

**Pull request merged**